gradle-jdks-integration-tests/src/jdkIntegrationTest/java/com/palantir/gradle/jdks/setup/GradleJdkInstallationSetupIntegrationTest.java

```diff
@@ -0,0 +1,286 @@
+    private static final String NON_EXISTING_CERT_ALIAS = "NonExistingCert";
+    private static final String CORRETTO_DISTRIBUTION_URL_ENV = "CORRETTO_DISTRIBUTION_URL";
+    private static final AmazonCorrettoJdkDistribution CORRETTO_JDK_DISTRIBUTION = new AmazonCorrettoJdkDistribution();
+    private static final List<String> NO_EXTRA_RUN_STEPS = List.of();
+    private static final List<String> INSTALL_CURL_RUN_STEP = List.of(getAptGetCurlRunStep());
+
+    private Path workingDir;
+
+    @BeforeEach
+    public void beforeEach() throws IOException {
+        workingDir = Files.createTempDirectory("testing-jdk-installation");
+    }
+
+    @AfterEach
+    public void afterEach() throws IOException {
+        FileUtils.deleteDirectory(workingDir.toFile());
+    }
+
+    @Test
+    public void can_setup_jdk_with_certs_centos() throws IOException, InterruptedException {
+        setupGradleDirectoryStructure(JDK_VERSION, Os.LINUX_GLIBC);
+        assertJdkWithNoCertsWasSetUp(dockerBuildAndRunTestingScript("centos:7", "/bin/bash", NO_EXTRA_RUN_STEPS));
+    }
+
+    @Test
+    public void can_setup_jdk_with_certs_ubuntu() throws IOException, InterruptedException {
+        setupGradleDirectoryStructure(JDK_VERSION, Os.LINUX_GLIBC);
+        assertJdkWithNoCertsWasSetUp(
+                dockerBuildAndRunTestingScript("ubuntu:20.04", "/bin/bash", INSTALL_CURL_RUN_STEP));
+    }
+
+    @Test
+    public void can_setup_jdk_with_certs_alpine() throws IOException, InterruptedException {
+        setupGradleDirectoryStructure(JDK_VERSION, Os.LINUX_MUSL);
+        assertJdkWithNoCertsWasSetUp(dockerBuildAndRunTestingScript("alpine:3.16.0", "/bin/sh", NO_EXTRA_RUN_STEPS));
+    }
+
+    @Test
+    public void can_setup_locally_from_scratch() throws IOException, InterruptedException {
+        Path temporaryGradleDirectory = setupGradleDirectoryStructure(JDK_VERSION, CurrentOs.get());
+        Path gradleHomeDir = Files.createTempDirectory("jdkIntegrationTest");
+        Path gradleJdksDir = Files.createDirectory(gradleHomeDir.resolve("gradle-jdks"));
+        Path expectedJavaHomeVersion =
+                gradleJdksDir.resolve(String.format("amazon-corretto-%s-%s", JDK_VERSION, TEST_HASH));
+        Path expectedJavaHome = expectedJavaHomeVersion.resolve("bin/java");
+
+        String firstRunOutput = runCommandWithZeroExitCode(
+                List.of(
+                        "/bin/bash",
+                        temporaryGradleDirectory
+                                .resolve("gradle-jdks-setup.sh")
+                                .toAbsolutePath()
+                                .toString()),
+                Map.of("GRADLE_USER_HOME", gradleHomeDir.toAbsolutePath().toString()));
+        assertThat(firstRunOutput).contains(String.format(SUCCESSFUL_OUTPUT + " %s", expectedJavaHomeVersion));
+        if (CaResources.readPalantirRootCaFromSystemTruststore(new StdLogger()).isPresent()) {
```

Comment:
For the tests we could probably just use a public root cert that's almost certainly in everyone's system truststores like one of [the Amazon root CAs](https://www.amazontrust.com/repository/)